cmd2/utils.py

```diff
@@ -470,10 +471,15 @@ def getbytes(self) -> bytes:
         """Get the internal contents as bytes"""
         return bytes(self.buffer.byte_buf)
 
-    def read(self) -> str:
+    def read(self, size: Optional[int] = -1) -> str:
         """Read from the internal contents as a str and then clear them out"""
-        result = self.getvalue()
-        self.clear()
+        if size is None or size == -1:
+            result = self.getvalue()
+            self.clear()
+        else:
+            result = self.buffer.byte_buf[:size].decode(encoding=self.encoding, errors=self.errors)
```

Comment:
God I love the new Codecov automatic comments showing you exactly where you need to add unit tests!!!

Reply:
Very strange. When I run locally it shows this is covered.